### PR TITLE
fix(quantic): fix issue with persisting search box suggestion value on blur

### PR DESCRIPTION
**Pull request by mmitiche**
## [SFINT-5665](https://coveord.atlassian.net/jira/software/c/projects/SFINT/boards/1076?assignee=62e7739dec6b328032f09736&selectedIssue=SFINT-5665)

The issue: 

- On blur and without submitting,  the value of the suggestion or the recent query was not persisted as the value of the search box input.
- On pressing the `Escape` key, the input content was automatically deleted.

https://github.com/user-attachments/assets/8f55f144-f445-4d6b-9d65-418d4ca6ef48

The solution:

- Updated the logic that handles the value of the input to allow persisting the values of the suggestions/recent queries before submitting them.
- prevented the default behaviour of the input on Escape key click.
- unit tests added.

https://github.com/user-attachments/assets/cb0ebc17-d34f-407e-b091-3f3571bb38a4

[SFINT-5665]: https://coveord.atlassian.net/browse/SFINT-5665?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/templates/defaultSearchBoxInput.html [hunk 1]
```diff
@@ -33,6 +32,7 @@
                 aria-haspopup="listbox"
                 aria-autocomplete="list"
                 data-cy="search-box-input"
+                data-value={inputValue}
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/templates/defaultSearchBoxInput.html:35]: I'm adding this to make sure the component renders when the value of this public property changes, 

I just discovered this in LWC, if a public property is not used in the HTML or in a getter, the component will not be reactive to it.

Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/templates/defaultSearchBoxInput.html:35]: Yea that's why they had the @track before.

**Review by erocheleau — APPROVED** [review]
I think it's adding more complexity, but it's good enough for now just to fix a quick bug.

Might need to be careful in the future to make sure these situations don't start to happen more and more frequently.

File: packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/templates/defaultSearchBoxInput.html (hunk 1 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/templates/defaultSearchBoxInput.html:35]: But  @track is not meant to be used with @api

**Pull request merged**